internal/internal_nexus_worker.go

```diff
@@ -0,0 +1,86 @@
+package internal
+
+import (
+	"github.com/nexus-rpc/sdk-go/nexus"
+	"go.temporal.io/api/workflowservice/v1"
+)
+
+type nexusWorkerOptions struct {
+	executionParameters workerExecutionParameters
+	client              Client
+	workflowService     workflowservice.WorkflowServiceClient
+	handler             nexus.Handler
+}
+
+type nexusWorker struct {
+	executionParameters workerExecutionParameters
+	workflowService     workflowservice.WorkflowServiceClient
+	worker              *baseWorker
+	stopC               chan struct{}
+}
+
+func newNexusWorker(opts nexusWorkerOptions) (*nexusWorker, error) {
+	workerStopChannel := make(chan struct{})
+	params := opts.executionParameters
+	params.WorkerStopChannel = getReadOnlyChannel(workerStopChannel)
+	ensureRequiredParams(&params)
+	poller := newNexusTaskPoller(
+		newNexusTaskHandler(
+			opts.handler,
+			opts.executionParameters.TaskQueue,
+			opts.client,
+			opts.executionParameters.DataConverter,
+			&workflowservice.RespondNexusTaskCompletedRequest{
+				Namespace: opts.executionParameters.Namespace,
+				Identity:  opts.executionParameters.Identity,
+			},
+			&workflowservice.RespondNexusTaskFailedRequest{
+				Namespace: opts.executionParameters.Namespace,
+				Identity:  opts.executionParameters.Identity,
+			},
```

Comment:
Can probably just pass the two fields in to build these requests instead of pre-building a request and cloning just for these two fields

Reply:
Yeah, I'll do that.